ibm/service/power/data_source_ibm_pi_volume.go

```diff
@@ -144,14 +165,22 @@ func dataSourceIBMPIVolumeRead(ctx context.Context, d *schema.ResourceData, meta
 	d.Set(Attr_AuxiliaryVolumeName, volumedata.AuxVolumeName)
 	d.Set(Attr_Bootable, volumedata.Bootable)
 	d.Set(Attr_ConsistencyGroupName, volumedata.ConsistencyGroupName)
+	d.Set(Attr_CreationDate, volumedata.CreationDate.String())
 	d.Set(Attr_DiskType, volumedata.DiskType)
+	if volumedata.FreezeTime != nil {
+		d.Set(Attr_FreezeTime, volumedata.FreezeTime.String())
+	} else {
+		d.Set(Attr_FreezeTime, nil)
+	}
```

Comment:
I believe the else part get automatically set to nil. Could double check on your local configuration?

Reply:
I added that if because otherwise it would throw an error due to a null pointer exception.

Reply:
Okay, I thought that would raise pointer exception if we did `d.Set(Attr_FreezeTime, volumedata.FreezeTime.String())` without checking if it was nil though.
@ismirlia  were you checking for nil first?

Reply:
@ismirlia Is this similar to what you had to do in the CLI?

Reply:
else block removed in latest commit.

Reply:
I don't remember the exact problem with the terraform, but I had an issue in the CLI where the returned freeze time was in the response nil. I wasn't sure if it needed to be set in the terraform.
```
                if volume.FreezeTime != nil {
			tbl.Add(common.HdrFreezeTime, volume.FreezeTime.String())
		}
```